overlays/haskell.nix

```diff
@@ -473,11 +473,14 @@ final: prev: {
                               { compiler.nix-name = args.compiler-nix-name; };
                   extra-hackages = args.extra-hackages or [];
                 };
-            in addProjectAndPackageAttrs {
-              inherit (pkg-set.config) hsPkgs;
-              inherit pkg-set;
-              plan-nix = callProjectResults.projectNix;
-              inherit (callProjectResults) index-state;
+                project = addProjectAndPackageAttrs rec {
+                  inherit (pkg-set.config) hsPkgs;
+                  inherit pkg-set;
+                  plan-nix = callProjectResults.projectNix;
+                  inherit (callProjectResults) index-state;
+                };
+            in project // {
+                projectCoverageReport = haskellLib.projectCoverageReport (haskellLib.selectProjectPackages project.hsPkgs);
```

Comment:
I'd be tempted to make people construct this call themselves, i.e. document `projectCoverageReport` and tell people to just pass in the packages they want coverage for. Maybe that's my inner Haskeller preferring functions over magic attributes, but it's harder to work out what to do in Nix land.

Reply:
Addressed these here: https://github.com/input-output-hk/haskell.nix/pull/762/commits/29ac8934f18dce96b7c8219577d5586117827808